packages/babel-helper-wrap-function/src/index.js

```diff
@@ -88,33 +98,18 @@ function plainFunction(path: NodePath, callId: Object) {
         done: false,
       },
     ).params,
-  }).expression;
-
-  if (isDeclaration && functionId) {
-    const declar = t.variableDeclaration("let", [
-      t.variableDeclarator(
-        t.identifier(functionId.name),
-        t.callExpression(container, []),
-      ),
-    ]);
-    (declar: any)._blockHoist = true;
-
-    if (path.parentPath.isExportDefaultDeclaration()) {
-      // change the path type so that replaceWith() does not wrap
-      // the identifier into an expressionStatement
-      path.parentPath.insertBefore(declar);
-      path.parentPath.replaceWith(
-        t.exportNamedDeclaration(null, [
-          t.exportSpecifier(
-            t.identifier(functionId.name),
-            t.identifier("default"),
-          ),
-        ]),
-      );
-      return;
-    }
+  });
 
-    path.replaceWith(declar);
+  if (isDeclaration) {
+    if (functionId) {
+      const basePath = path.parentPath.isExportDeclaration()
```

Comment:
seems like the same case as in https://github.com/babel/babel/pull/6963#discussion_r154773120

a helper or a method could be created for future reuse